src/common/src/types/interval.rs

```diff
@@ -66,6 +67,12 @@ impl IntervalUnit {
         self.ms
     }
 
+    pub fn get_total_ms(&self) -> i64 {
+        self.ms
+            + (self.days as i64) * 24 * 3600 * 1000
+            + (self.months as i64) * 30 * 24 * 3600 * 1000
+    }
+
```

Comment:
`IntervalUnit` uses a (month, day, ms) tuple because it is not always possible to convert to `ms`: each month has different number of days, and a day can have 23 or 25 hours during DST adjustment. ([ref](https://www.postgresql.org/docs/current/datatype-datetime.html#:~:text=Internally%20interval%20values%20are%20stored%20as%20months%2C%20days%2C%20and%20microseconds.%20This%20is%20done%20because%20the%20number%20of%20days%20in%20a%20month%20varies%2C%20and%20a%20day%20can%20have%2023%20or%2025%20hours%20if%20a%20daylight%20savings%20time%20adjustment%20is%20involved.))

In PostgreSQL:
```
test=# select * from generate_series(timestamp '2020-01-01 01:02:03', timestamp '2021-01-01 03:04:05', interval '1' month);
   generate_series   
---------------------
 2020-01-01 01:02:03
 2020-02-01 01:02:03
 2020-03-01 01:02:03
 2020-04-01 01:02:03
 2020-05-01 01:02:03
 2020-06-01 01:02:03
 2020-07-01 01:02:03
 2020-08-01 01:02:03
 2020-09-01 01:02:03
 2020-10-01 01:02:03
 2020-11-01 01:02:03
 2020-12-01 01:02:03
 2021-01-01 01:02:03
(13 rows)
```

Reply:
Yes, I think the best way is to use DataChunkBuilder introduced in another PR. Will do it later